### DISCUSSION TITLE
MultiSelect - Show Toggle All checkbox when having selectionLimit

### DISCUSSION THREAD

**Discussion by ghost** [primefaces/community#2162] [PrimeNG]
I checked the code of the `MultiSelect` component and I see that we are hiding the `toggleAll` checkbox when having `selectionLimit`.

You can see the logic [here](https://github.com/primefaces/primeng/blob/1a7c2cf489e3d34e99661c3e04cf79685803d519/src/app/components/multiselect/multiselect.ts#L269) and [here](https://github.com/primefaces/primeng/blob/1a7c2cf489e3d34e99661c3e04cf79685803d519/src/app/components/multiselect/multiselect.ts#L1466)

In our project we need to have the `toggleAll` checkbox. Imagine this situation: You have a `selectionLimit` of 40 and you want to select/unselect all of them.

Is there any historical reason that explains why we are hiding the `toggleAll` checkbox when passing `selectionLimit`? Can we enable it somehow?

**Comment by dennis6101990leon** [comment]
Hello,

Hiding "toogleAll" with selectionLimit prevents confusion and potential [HealthCareGov](https://www.health-insurancemarket.com) performance issues. You can enable it in two ways: Check your library's configuration options like showToggleAll in PrimeNG. 

If no option exists, implement a custom solution by overriding the template and handling the checkbox click to select/deselect within the limit.

**Reply by ghost** [reply]
Hi @dennis6101990leon thanks for your answer. `showToggleAll` is true by default for the MultiSelect component. How can I override the template and add a custom checkbox? Do you have an example?

Thanks!
